### PR TITLE
Added count_and_say.py

### LINKED ISSUE #9908: [ADD]: Combination_Sum II
### Feature description

The count-and-say sequence is a series of number strings that follows a recursive formula. The first number in the sequence is "1". 

To get the next number in the sequence, you "say" the previous number and then convert it into a new number string.

 To "say" a number string, you need to split it into the smallest number of substrings possible, where each substring has only one unique digit. For each of these substrings, you say the number of digits in it and then the digit itself. 

Finally, you concatenate all the said digits. For instance, if the number string is "3322251", here's how you would "say" it and convert it.

Web capture_6-10-2023_15625_leetcode com

### PR DESCRIPTION
**Pull request by theinit01**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #9908 ".

### CODE REVIEW

File: strings/count_and_say.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+def countandsay(n: int) -> str:
```
[the remark this answers is not in the document]
Comment (theinit01) [reply strings/count_and_say.py]: Changed the name n to number

**Comment by theinit01** [comment]
issue #9908 had the name ADD: Combination Sum 2, but contained count and say algorithm. So I added count and say algorithm.

File: backtracking/generate_parentheses.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+def generateparenthesis(n):
```
[the remark this answers is not in the document]
Comment (theinit01) [reply backtracking/generate_parentheses.py]: Added descriptive name for n
Added type hints

File: backtracking/generate_parentheses.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        ['()']
+
+        >>> generateparenthesis(2)
+        ['(())', '()()']
+
+        >>> generateparenthesis(3)
+        ['((()))', '(()())', '(())()', '()(())', '()()()']
+    """
+
+    def backtrack(result, current, open_count, close_count, max_pairs):
```
[the remark this answers is not in the document]
Comment (theinit01) [reply backtracking/generate_parentheses.py]: Added doctests for backtrack
Added Type hints

File: backtracking/generate_parentheses.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
+def generateparenthesis(number):
```
[the remark this answers is not in the document]
Comment (theinit01) [reply backtracking/generate_parentheses.py]: added type hints for generateparenthesis and number

File: backtracking/generate_parentheses.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        ['()']
+
+        >>> generateparenthesis(2)
+        ['(())', '()()']
+
+        >>> generateparenthesis(3)
+        ['((()))', '(()())', '(())()', '()(())', '()()()']
+    """
+
+    def backtrack(result, current, open_count, close_count, max_pairs):
```
[the remark this answers is not in the document]
Comment (theinit01) [reply backtracking/generate_parentheses.py]: Added doctests and added type hints

File: backtracking/generate_parentheses.py [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+            result.append(current)
+            return
+
+        if open_count < max_pairs:
+            backtrack(result, current + "(", open_count + 1, close_count, max_pairs)
+
+        if close_count < open_count:
+            backtrack(result, current + ")", open_count, close_count + 1, max_pairs)
+
+    result = []
```
Comment (haxkd) [inline backtracking/generate_parentheses.py:63]: describe it, the result  type

**Pull request closed without merging**